common/src/main/java/io/netty/util/internal/logging/InternalLoggerFactory.java

```diff
@@ -31,7 +33,11 @@
  * as possible and shouldn't be called more than once.
  */
 public abstract class InternalLoggerFactory {
-    private static volatile InternalLoggerFactory defaultFactory;
+
+    private static class InternalLoggerHolder {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a private constructor to hide the implicit public one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1118)